src/libcore/iter.rs

```diff
@@ -369,6 +369,25 @@ pub trait Iterator {
     /// `None` here means that either there is no known upper bound, or the
     /// upper bound is larger than `usize`.
     ///
+    /// # Safety
```

Comment:
This is in some sense a meaningless section, as this is true for literally all traits. You can't trust it to do a damn thing, other than do "something" memory-safe.